sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -348,14 +327,16 @@ public void AddLink(string traceparent, string? tracestate, IDictionary<string,
                     _currentActivity.Start();
                 }
 
+                _isAllDataRequested = _currentActivity?.GetIsAllDataRequestedFlag() ?? true;
```

Comment:
```suggestion
```

Reply:
in the line 270, right after starting activity.The it'd apply to Activity source path and would immediately prevent `_currentActivity.AddTag(OpenTelemetrySchemaAttribute, OpenTelemetrySchemaVersion);` from adding any overhead